pynsights/cli.py

```diff
@@ -0,0 +1,124 @@
+# - When you run `python -m pynsights` python will execute
+#   `__main__.py` as a script. That means there won't be any
+#   `pynsights.__main__` in `sys.modules`.
+# - When you import `__main__` it will get executed again (as a module) because
+#   there's no `pynsights.__main__` in `sys.modules`.
+
+"""Module that contains the command line application."""
+
+from __future__ import annotations
+
+import argparse
+import sys
+import tempfile
+from contextlib import suppress
+from pathlib import Path
+from runpy import run_module, run_path
+
+from pynsights import Recorder
+from pynsights.view import view
+
+
+def get_parser() -> argparse.ArgumentParser:
+    """
+    Return the program argument parser.
+
+    Returns:
+        The argument parser for the program.
+    """
+    usage = "%(prog)s [GLOBAL_OPTS...] COMMAND [COMMAND_OPTS...]"
+    description = "Understand Python programs by visualizing how modules interact."
+    parser = argparse.ArgumentParser(add_help=False, usage=usage, description=description, prog="pynsights")
+
+    main_help = "Show this help message and exit. Commands also accept the -h/--help option."
+    command_help = "Show this help message and exit."
+
+    global_options = parser.add_argument_group(title="Global options")
+    global_options.add_argument("-h", "--help", action="help", help=main_help)
+
+    subparsers = parser.add_subparsers(dest="command", title="Commands", metavar="", prog="pynsights")
+
+    def subparser(command: str, text: str, **kwargs) -> argparse.ArgumentParser:
+        sub = subparsers.add_parser(command, add_help=False, help=text, description=text, **kwargs)
+        sub.add_argument("-h", "--help", action="help", help=command_help)
+        return sub
+
+    record_parser = subparser("record", "Record the interactions within the execution of a Python program.")
+    record_parser.add_argument("-o", "--output", type=Path, default=None, help="Output path of the recorded data.")
+    record_parser.add_argument("record_command", nargs="+", help="The Python script/module to run and its options.")
+
+    render_parser = subparser("render", "Render the given data file into a standalone HTML file.")
+    render_parser.add_argument("-w", "--browser", "--open", dest="browser", action="store_true", help="Open the HTML file in your default browser.")
+    render_parser.add_argument("input_file", type=Path, help="The input data file to render.")
+    render_parser.add_argument("-o", "--output", type=Path, default=None, help="Output path of the standalone HTML.")
+
+    run_parser = subparser("run", "Record and render in one go.")
+    run_parser.add_argument("-w", "--browser", "--open", dest="browser", action="store_true", help="Open the HTML file in your default browser.")
+    run_parser.add_argument("-o", "--output", type=Path, default=None, help="Output path of the final HTML file.")
+    run_parser.add_argument("record_command", nargs="+", help="The Python script/module to run and its options.")
```

Comment:
Command names and options can absolutely be changed. Please let me know if you want to rename or change something (don't hesitate, it's really easy to change).